

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  

  <title>Immune receptor (IR) model &mdash; scirpy  documentation</title>
  


  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true, "ignoreClass": "document", "processClass": "math|output_area"}})</script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/scanpy.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/typehints.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Glossary" href="glossary.html" />
    <link rel="prev" title="scirpy.ir_dist.metrics.AlignmentDistanceCalculator.squarify" href="generated/scirpy.ir_dist.metrics.AlignmentDistanceCalculator.squarify.html" />
 


</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/scirpy_logo_bright.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                0.7.1.dev6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="usage-principles.html">Usage principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="api.html">API</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Immune receptor (IR) model</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#what-is-a-clonotype">What is a clonotype?</a></li>
<li class="toctree-l2"><a class="reference internal" href="#dual-tcrs-and-allelically-included-b-cells">Dual TCRs and allelically included B cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="#clonotype-definition">Clonotype definition</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="glossary.html">Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="versions.html">Other versions</a></li>
<li class="toctree-l1"><a class="reference internal" href="zzz_bibliography.html">Bibliography</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">scirpy</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Immune receptor (IR) model</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            
              <a href="https://github.com/icbi-lab/scirpy/blob/master/docs/ir-biology.rst" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="immune-receptor-ir-model">
<span id="receptor-model"></span><h1>Immune receptor (IR) model<a class="headerlink" href="#immune-receptor-ir-model" title="Permalink to this headline">¶</a></h1>
<section id="what-is-a-clonotype">
<h2>What is a clonotype?<a class="headerlink" href="#what-is-a-clonotype" title="Permalink to this headline">¶</a></h2>
<p>A clonotype designates a collection of T or B cells that descend from a
common, antecedent cell, and therefore, bear the same adaptive
immune receptors (<a class="reference internal" href="glossary.html#term-IR"><span class="xref std std-term">IR</span></a>) and recognize the same <a class="reference internal" href="glossary.html#term-Epitope"><span class="xref std std-term">epitopes</span></a>.</p>
<p>In single-cell RNA-sequencing (scRNA-seq) data, T or B cells sharing identical
complementarity-determining regions 3 (<a class="reference internal" href="glossary.html#term-CDR3"><span class="xref std std-term">CDR3</span></a>) nucleotide sequences of both
<a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VJ</span></a> and <a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VDJ</span></a> chains (e.g. both α and β <a class="reference internal" href="glossary.html#term-TCR"><span class="xref std std-term">TCR</span></a> chains)
make up a clonotype. Scirpy provides an option to additionally
require clonotypes to have the same <a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">V-gene</span></a>, enforcing the CDR 1
and 2 regions to be the same.</p>
</section>
<section id="dual-tcrs-and-allelically-included-b-cells">
<h2>Dual TCRs and allelically included B cells<a class="headerlink" href="#dual-tcrs-and-allelically-included-b-cells" title="Permalink to this headline">¶</a></h2>
<p>Contrary to what would be expected based on the previously described
mechanism of allelic exclusion (<span id="id1">[<a class="reference internal" href="zzz_bibliography.html#id7" title="Brenna L Brady, Natalie C Steinel, and Craig H Bassing. Antigen receptor allelic exclusion: an update and reappraisal. J. Immunol., 185(7):3801–3808, October 2010.">BSB10</a>]</span>), scRNA-seq datasets can
feature a considerable number of cells with more than one pair of
<a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VJ</span></a> and <a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VDJ</span></a> chains. Since cells with more than
one productive CDR3 sequence for each chain did not fit into our
understanding of adaptive immune cell biology, most TCR analysis tools ignore
these cells (<span id="id2">[<a class="reference internal" href="zzz_bibliography.html#id9" title="David S. Fischer, Yihan Wu, Benjamin Schubert, and Fabian J. Theis. Predicting antigen-specificity of single t-cells based on TCR CDR3 regions. BioRxiv, August 2019. URL: https://doi.org/10.1101/734053, doi:10.1101/734053.">FWST19</a>]</span>, <span id="id3">[<a class="reference internal" href="zzz_bibliography.html#id10" title="Lei Zhang, Xin Yu, Liangtao Zheng, Yuanyuan Zhang, Yansen Li, Qiao Fang, Ranran Gao, Boxi Kang, Qiming Zhang, Julie Y Huang, Hiroyasu Konno, Xinyi Guo, Yingjiang Ye, Songyuan Gao, Shan Wang, Xueda Hu, Xianwen Ren, Zhanlong Shen, Wenjun Ouyang, and Zemin Zhang. Lineage tracking reveals dynamic relationships of T cells in colorectal cancer. Nature, 564(7735):268–272, December 2018.">ZYZ+18</a>]</span>), or select the CDR3
sequence with the highest expression level (<span id="id4">[<a class="reference internal" href="zzz_bibliography.html#id8" title="Shaked Afik, Kathleen B Yates, Kevin Bi, Samuel Darko, Jernej Godec, Ulrike Gerdemann, Leo Swadling, Daniel C Douek, Paul Klenerman, Eleanor J Barnes, Arlene H Sharpe, W Nicholas Haining, and Nir Yosef. Targeted reconstruction of T cell receptor sequence from single cell RNA-seq links CDR3 length to T cell differentiation state. Nucleic Acids Res., 45(16):e148, September 2017.">AYB+17</a>]</span>). While in
some cases these cells might represent artifacts (e.g. doublets of a CD8+ and
a CD4+ T cell engaged in an immunological synapse), there is an increasing
amount of evidence in support of a bone fide <a class="reference internal" href="glossary.html#term-Dual-IR"><span class="xref std std-term">Dual IR</span></a> population
(<span id="id5">[<a class="reference internal" href="zzz_bibliography.html#id5" title="Nathaniel J. Schuldt and Bryce A. Binstadt. Dual TCR t cells: identity crisis or multitaskers? The Journal of Immunology, 202(3):637–644, January 2019. URL: https://doi.org/10.4049/jimmunol.1800904, doi:10.4049/jimmunol.1800904.">SB19</a>]</span>, <span id="id6">[<a class="reference internal" href="zzz_bibliography.html#id11" title="Qingyong Ji, Antoine Perchellet, and Joan M Goverman. Viral infection triggers central nervous system autoimmunity via activation of CD8+ T cells expressing dual TCRs. Nat. Immunol., 11(7):628–634, July 2010.">JPG10</a>]</span>, <span id="id7">[<a class="reference internal" href="zzz_bibliography.html#id3" title="Christian Vettermann and Mark S. Schlissel. Allelic exclusion of immunoglobulin genes: models and mechanisms. Immunological Reviews, 237(1):22–42, August 2010. URL: https://doi.org/10.1111/j.1600-065x.2010.00935.x, doi:10.1111/j.1600-065x.2010.00935.x.">VS10</a>]</span>).</p>
<p>Scirpy allows investigating the composition and phenotypes of both single-
and dual-IR cells by leveraging a immune cell model similar to the one
proposed in <span id="id8">[<a class="reference internal" href="zzz_bibliography.html#id12" title="Michael J T Stubbington, Tapio Lönnberg, Valentina Proserpio, Simon Clare, Anneliese O Speak, Gordon Dougan, and Sarah A Teichmann. T cell fate and clonality inference from single-cell transcriptomes. Nat. Methods, 13(4):329–332, April 2016.">SLonnbergP+16</a>]</span>, where immune cells are allowed to
have a primary and a secondary pair of <a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VJ</span></a> and
<a class="reference internal" href="glossary.html#term-V-D-J"><span class="xref std std-term">VDJ</span></a> chains. For each cell, the primary pair consists of the VJ-
and VDJ-chain with the highest read count. Likewise, the secondary pair is the
pair of VJ/VDJ-chains with the second highest expression level. Based on the
assumption that each cell has only two copies of the underlying chromosome
set, if more than two variants of a chain are recovered for the same cell,
the excess IR chains are ignored by Scirpy and the corresponding cells
flagged as <a class="reference internal" href="glossary.html#term-Multichain-cell"><span class="xref std std-term">Multichain-cell</span></a>. This filtering strategy leaves the choice
of discarding or including multichain cells in downstream analyses.</p>
</section>
<section id="clonotype-definition">
<h2>Clonotype definition<a class="headerlink" href="#clonotype-definition" title="Permalink to this headline">¶</a></h2>
<p>Scirpy implements a network-based clonotype definition that enables
clustering cells into <a class="reference internal" href="glossary.html#term-Clonotype"><span class="xref std std-term">clonotypes</span></a> based on <em>identical</em>
<a class="reference internal" href="glossary.html#term-CDR3"><span class="xref std std-term">CDR3</span></a> <em>nucleotide sequences</em> and into <a class="reference internal" href="glossary.html#term-Clonotype-cluster"><span class="xref std std-term">clonotype clusters</span></a>
based on</p>
<ul class="simple">
<li><p><em>identical CDR3 amino acid sequences</em>, or</p></li>
<li><p><em>similar CDR3 amino acid sequences</em> based on pairwise sequence alignment.</p></li>
</ul>
<p>The latter approach is inspired by studies showing that similar TCR sequences
also share epitope targets (<span id="id9">[<a class="reference internal" href="zzz_bibliography.html#id9" title="David S. Fischer, Yihan Wu, Benjamin Schubert, and Fabian J. Theis. Predicting antigen-specificity of single t-cells based on TCR CDR3 regions. BioRxiv, August 2019. URL: https://doi.org/10.1101/734053, doi:10.1101/734053.">FWST19</a>]</span>, <span id="id10">[<a class="reference internal" href="zzz_bibliography.html#id13" title="Jacob Glanville, Huang Huang, Allison Nau, Olivia Hatton, Lisa E Wagar, Florian Rubelt, Xuhuai Ji, Arnold Han, Sheri M Krams, Christina Pettus, Nikhil Haas, Cecilia S Lindestam Arlehamn, Alessandro Sette, Scott D Boyd, Thomas J Scriba, Olivia M Martinez, and Mark M Davis. Identifying specificity groups in the T cell receptor repertoire. Nature, 547(7661):94–98, July 2017.">GHN+17</a>]</span>,
<span id="id11">[<a class="reference internal" href="zzz_bibliography.html#id6" title="Pradyot Dash, Andrew J. Fiore-Gartland, Tomer Hertz, George C. Wang, Shalini Sharma, Aisha Souquette, Jeremy Chase Crawford, E. Bridie Clemens, Thi H. O. Nguyen, Katherine Kedzierska, Nicole L. La Gruta, Philip Bradley, and Paul G. Thomas. Quantifiable predictive features define epitope-specific t cell receptor repertoires. Nature, 547(7661):89–93, June 2017. URL: https://doi.org/10.1038/nature22383, doi:10.1038/nature22383.">DFGH+17</a>]</span>). Based on these approaches, Scirpy constructs a global,
epitope-focused cell similarity network. While convergence of the
nucleotide-based clonotype definition to the amino acid based one hints at
selection pressure, sequence alignment based networks offer the opportunity
to identify cells that might recognize the same epitopes.</p>
</section>
</section>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="glossary.html" class="btn btn-neutral float-right" title="Glossary" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="generated/scirpy.ir_dist.metrics.AlignmentDistanceCalculator.squarify.html" class="btn btn-neutral float-left" title="scirpy.ir_dist.metrics.AlignmentDistanceCalculator.squarify" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2021, Gregor Sturm, Tamas Szabo.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>